3rd_party/catch2/catch2_main.cpp

```diff
@@ -0,0 +1,3 @@
+#define CATCH_CONFIG_MAIN
+
+#include "opentelemetry/3rd_party/catch.hpp"
```

Comment:
I don't see the main function, do I miss anything? If it is based on the macro you can just do it in the build rule to set the macro.

Reply:
It's from the macro, yes.

You might have a situation like this
```
// a_test_helper.cc
#include "opentelemetry/3rd_party/catch.hpp"
... <define some helper functions>

// a_test.cc
#include "a_test_helper.h"
#include "opentelemetry/3rd_party/catch.hpp"
... <define tests>
```
If the macro definition gets propagated to both source files when they get compiled, you'll have two mains.